nomic/project.py

```diff
@@ -0,0 +1,1370 @@
+
+        * **project_name** - The name of the project.
+        * **description** - A description for the project.
+        * **unique_id_field** - The field that uniquely identifies each datum. If a datum does not contain this field, it will be added and assigned a random unique ID.
+        * **modality** - The data modality of this project. Currently, Atlas supports either `text` or `embedding` modality projects.
+        * **organization_name** - The name of the organization to create this project under. You must be a member of the organization with appropriate permissions. If not specified, defaults to your user account's default organization.
+        * **is_public** - Should this project be publicly accessible for viewing (read only). If False, only members of your Nomic organization can view.
+        * **reset_project_if_exists** - If the requested project exists in your organization, will delete it and re-create it.
+        * **project_id** - An alternative way to retrieve a project is by passing the project_id directly. This only works if a project exists.
+        * **reset_project_if_exists** - If the specified project exists in your organization, reset it by deleting all of its data. This means your uploaded data will not be contextualized with existing data.
+        * **add_datums_if_exists** - If specifying an existing project and you want to add data to it, set this to true.
+        **Returns:** project_id on success.
+
+        """
+        assert name is not None or project_id is not None, "You must pass a name or project_id"
+
+        super().__init__()
+
+        if project_id is not None:
+            self.meta = self._get_project_by_id(project_id)
+            return
+
+
+        results = self._get_existing_project_by_name(project_name=name, organization_name=organization_name)
+        organization_name = results['organization_name']
+
+        if 'project_id' in results: #project already exists
+            project_id = results['project_id']
+            if reset_project_if_exists:
+                logger.info(
+                    f"Found existing project `{name}` in organization `{organization_name}`. Clearing it of data by request."
+                )
+                self._delete_project_by_id(project_id=project_id)
+                project_id = None
+            else:
+                if add_datums_if_exists:
+                    logger.info(
+                        f"Using existing project `{name}` in organization `{organization_name}`."
+                    )
+                else:
+                    raise ValueError(
+                        f"Project already exists with the name `{name}` in organization `{organization_name}`. "
+                        f"You can add datums to it by settings `add_datums_if_exists = True` or reset it by specifying `reset_project_if_exist=True` on a new upload."
+                    )
+
+
+        if project_id is None: #if there is no existing project, make a new one.
```

Comment:
can you document this in the docstring pls? like just specify when this method will create new vs use old